lib/spack/spack/solver/asp.py

```diff
@@ -726,10 +726,17 @@ def pkg_rules(self, pkg, tests):
     def package_dependencies_rules(self, pkg, tests):
         """Translate 'depends_on' directives into ASP logic."""
         for name, conditions in sorted(pkg.dependencies.items()):
-            for cond, dep in sorted(conditions.items()):
+            for cond_id, (cond, dep) in enumerate(sorted(conditions.items())):
```

Comment:
Not related to this PR, but maybe it's worth rewriting the outer loop as:
```python
for _, conditions in sorted(pkg.dependencies.items()):
```
to clarify that the key is used only for sorting.